spinalcordtoolbox/utils/fs.py

```diff
@@ -47,8 +52,18 @@ def close(self):
             self.fd2.close()
 
     def write(self, text):
-        self.fd1.write(text)
-        self.fd2.write(text)
+        # Quick and dirty solution to filter out ansi escape sequences (like color codes) from
+        # the text if one file is an interactive terminal but the other one is not.
+        # This will only filter out cases where the entire escape sequence is part of a single
+        # call to write(). See https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4287
+        if self.fd2.isatty() and not self.fd1.isatty():
+            self.fd1.write(ansi_escape.sub('', text))
```

Comment:
From what I can tell, this _does_ work for me!

Terminal output:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/37a49438-6d6a-4433-ab83-61a6eb3f0fa8)

Output to `sct_run_batch_log.txt` (`master`):

```
Your batch completed but some subjects may have not completed successfully, please consult the logs for:
sub-01
sub-03
sub-05

Started: 2023-11-07 15h07m13s | Ended: 15h07m13s | Duration: 00h00m00s

[0m
To open the Quality Control (QC) report in a web-browser, type:[0m
[92mxdg-open /home/joshua/repos/spinalcordtoolbox/data/sct_course_data/multi_subject/output/qc/index.html
[0m
```

Output to `sct_run_batch_log.txt` (`mgp/tee-quick-fix`):

```
Your batch completed but some subjects may have not completed successfully, please consult the logs for:
sub-01
sub-03
sub-05

Started: 2023-11-07 15h04m53s | Ended: 15h04m54s | Duration: 00h00m00s


To open the Quality Control (QC) report in a web-browser, type:
xdg-open /home/joshua/repos/spinalcordtoolbox/data/sct_course_data/multi_subject/output/qc/index.html
```